src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1422,6 +1435,16 @@ void StorageReplicatedMergeTree::checkParts(bool skip_sanity_checks)
             continue;
         }
 
+        {
+            std::lock_guard lock(outdated_data_parts_mutex);
```

Comment:
It's very unclear that we are going to retry only when we have unexpected uncovered parts. Add a comment.